tests/integrations/mcs/tso/server_test.go

```diff
@@ -578,3 +585,212 @@ func (suite *CommonTestSuite) TestBootstrapDefaultKeyspaceGroup() {
 	suite.pdLeader.ResignLeader()
 	suite.pdLeader = suite.cluster.GetServer(suite.cluster.WaitLeader())
 }
+
+func TestTSOServiceSwitch1(t *testing.T) {
+	re := require.New(t)
+	re.NoError(failpoint.Enable("github.com/tikv/pd/client/fastUpdateServiceMode", `return(true)`))
+	var wg sync.WaitGroup
+	defer wg.Wait()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	tc, err := tests.NewTestAPICluster(ctx, 1,
+		func(conf *config.Config, _ string) {
+			conf.MicroService.EnableTSOFallback = true
+		},
+	)
+	re.NoError(err)
+	defer tc.Destroy()
+
+	err = tc.RunInitialServers()
+	re.NoError(err)
+
+	leaderName := tc.WaitLeader()
+	re.NotEmpty(leaderName)
+	pdLeader := tc.GetServer(leaderName)
+	backendEndpoints := pdLeader.GetAddr()
+	re.NoError(pdLeader.BootstrapCluster())
+	pdClient, err := pd.NewClientWithContext(ctx, []string{backendEndpoints}, pd.SecurityOption{})
+	re.NoError(err)
+	defer pdClient.Close()
+	ch := make(chan struct{})
+	ch1 := make(chan struct{})
+	wg.Add(1)
+	go func(ctx context.Context, wg *sync.WaitGroup, ch, ch1 chan struct{}) {
+		defer wg.Done()
+		var lastPhysical, lastLogical int64
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			default:
+			}
+			physical, logical, err := pdClient.GetTS(context.Background())
+			if err == nil {
+				re.GreaterOrEqual(physical, lastPhysical)
+				if physical == lastPhysical {
+					re.Greater(logical, lastLogical)
+				}
+				lastPhysical = physical
+				lastLogical = logical
+				select {
+				case <-ch1:
+					ch <- struct{}{}
+				default:
+				}
+			} else {
+				t.Log(err)
+			}
```

Comment:
Prefer to wrap a function

```suggestion
    waitOneTS := func() {
        ch1 <- struct{}{}
	    <-ch
    }
	...
    waitOneTS()
```